reco_utils/dataset/wikidata.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import requests
+
+def find_wikidataID(name):
+    """Find the entity ID in wikidata from a title string.
+    Args:
+        name (str): A string with search terms (eg. "Batman (1989) film")
+    Returns:
+        entityID: wikidata entityID corresponding to the title string. 
+                  'entityNotFound' will be returned if no page is found
+    """
+    r = requests.get("https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch="+name+"&format=json&prop=pageprops&ppprop=wikibase_item")
```

Comment:
do you mind to explain what are these requests doing, and why are you chaining them?

Reply:
Sure! In the previous version I was matching a string (the name of the movie) with the title of a wikipedia page, this was an issue for ambiguous names (eg. Brazil movie vs. Brazil the country). I added a previous request that receives a free text string and returns the first page found in a wikipedia, for Movielens I send "MovieTitle, MovieYear, film" and I then use the response with the wikipedia page title to query Wikidata